samples/bootloader/src/main.c

```diff
@@ -189,6 +189,23 @@ void main(void)
 	}
 #endif /* CONFIG_SOC_NRF9160 */
 
-	boot_from((u32_t *)s0_address_read());
+	const struct fw_firmware_info *s0_fw_info = fw_firmware_info_get(s0_address_read());
+	const struct fw_firmware_info *s1_fw_info = fw_firmware_info_get(s1_address_read());
```

Comment:
Your patch breaks 80 columns, check it :)

Reply:
Perhaps introduce some helper variables simply to make the code more readable.

```
u32_t s0_addr = s0_address_read();
u32_t s1_addr = s1_address_read();
const struct s0 = fw_firmware_info_get(s0_addr);
const struct s1 = fw_firmware_info_get(s1_addr)

if (s0 != NULL && s1 != NULL) {
    if (s0->firmware_version >= s1->firmware_version) {
        boot_from(s0_addr);
     }
}
```

etc. up to you :)

Reply:
Did the refactoring you suggested to reduce the 80 columns ^^